mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -4596,3 +4686,28 @@ func BinaryAuthorizationDiffSuppress(k, old, new string, r *schema.ResourceData)
 
 	return false
 }
+
+<% unless version == 'ga' -%>
+func containerClusterNodePoolAutoConfigDiffSuppress(k, old, new string, r *schema.ResourceData) bool {
+	// emptyOrUnsetBlockDiffSuppress treats any resource arguments with type schema.TypeList under the
+	// desired config block as another config block, and panics when []interface{} cannot be asserted
+	// as []map[string]interface{}. Run it only against the config block itself.
+	if k == "node_pool_auto_config.#" {
+		return emptyOrUnsetBlockDiffSuppress(k, old, new, r)
+	}
+	return false
+}
+
+func validateNodePoolAutoConfig(cluster *container.Cluster) error {
```

Comment:
could you add unit tests for this function?

Reply:
Whoops, had a copy/paste typo in there. Fixed and test added.